services/src/services/update/update.class.js

```diff
@@ -0,0 +1,177 @@
+const FeathersSequelize = require('feathers-sequelize');
+
+/*
+ * This class exist mostly as a wrapper around the feathers-sequelize service
+ * for the Update model.
+ */
+class Update extends FeathersSequelize.Service {
+  constructor(options) {
+    super(options);
+
+    this.app = options.app;
+
+    /*
+     * Undefining some APIs which we don't want/need.
+     */
+    this.update = undefined;
+    this.patch = undefined;
+    this.remove = undefined;
+  }
+
+  async get(id, params) {
+    const instance = await this.app.service('status').get(id);
+    let latestVersion = await this.app.service('versions').find({
+      query: {
+        uuid: id,
+        $limit: 1
+      },
+    });
+
+    let findParams = {
+      query: params.query, /* copy the original query parameters over */
+    };
+    this.scopeFindQuery(findParams);
+
+    return this.find(findParams).then((records) => {
+      let record = records[0];
+      let computed = {
+        meta: {
+          channel: record.channel,
+          level: record.id,
+        },
+        core: record.manifest.core,
+        plugins: {
+          updates: [
+          ],
+          deletes: [
+          ],
+        },
+      };
+
+      this.prepareManifestFromRecord(record, computed);
+      /*
+       * Last but not least, make sure that any flavor specific updates are
+       * assigned to the computed update manifest
+       */
+      this.prepareManifestWithFlavor(instance, record, computed);
+
+      if (latestVersion.length === 1) {
```

Comment:
Because of the `$limit` above, I guess this `if` shouldn't even be here because it will always be true right?

Reply:
OK, there could also be 0. But is that really even a possibility? If so I guess yes a `logger.something` could be very useful, even more at that stage of the project.

Reply:
A latestVersion of zero is a possibility for a fresh installation